### PR TITLE
feat(collator): add check value flow on finalize

### PR DESCRIPTION
**Pull request by serejkaaa512**

### CODE REVIEW

**Review by 0xdeafbeef — APPROVED** [review]
lgtm

**Comment by SmaGMan** [comment]
Should run the check by param in node config. Default should be off.

File: collator/src/collator/do_collate.rs [hunk 1]
```diff
@@ -246,6 +247,7 @@ impl CollatorStdImpl {
                     finalize_params,
                     prepare_groups_wu_total,
                     execute_groups_wu_total,
+                    check_value_flow,
```
Comment (SmaGMan) [inline collator/src/collator/do_collate.rs:250]: Should pass full `CollatorConfig` here

**Review by SmaGMan — CHANGES_REQUESTED** [review]
And squash commits, pls.

**Review by SmaGMan — APPROVED** [review]

**Pull request merged**